shell/server/installer.js

```diff
@@ -102,6 +128,28 @@ cancelDownload = function (packageId) {
   }
 }
 
+if (!Meteor.settings.replicaNumber) {
+  installers = {};
+
+  Meteor.startup(function () {
+    // Restart any deletions that were killed while in-progress.
+    Packages.find({status: "delete"}).forEach(function (pkg) {
+      deletePackageInternal(pkg._id);
+    });
+
+    // Watch for new installation requests and fulfill them.
+    Packages.find({status: {$in: ["download", "verify", "unpack", "analyze"]}}).observe({
+      added: startInstallInternal,
+      removed: cancelDownloadInternal
+    });
+
+    // Watch for new cleanup requests and fulfill them.
+    Packages.find({status: "ready", shouldCleanup: true}).observe({
+      added: deletePackageInternal
```

Comment:
I think this needs to be `changed`, not `added`, to work as you intend. You want this to be called immediately after any replica calls `deletePackage()`, right?